### PR TITLE
CI: only run backend lints once, not in all test environments

### PR DESCRIPTION
**Pull request by akx**
# Description

No need to spend cycles running lints multiple times; just run it once in a separate step.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abidlabs** [comment]
Hi @akx thanks yet again! This change looks fine to me, but let's run linting before we run the tests. Linting is much faster than running the tests, so better for it to fail quickly if the code is not linted rather than having to wait for all of the tests to run

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -56,7 +56,28 @@ jobs:
     - name: Tests
       shell: bash
       run: |
-        bash scripts/ci.sh
+        bash scripts/test.sh
+  client-lint:
```
Comment (freddyaboulton) [inline .github/workflows/backend.yml:60]: Thanks for the PR @akx . I think `needs` is the right approach here: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

That way the tests don't run if lint fails

**Comment by akx** [comment]
@abidlabs @freddyaboulton Done deal. `client-test` needs `client-lint`, `test` needs `lint` now.

**Review by freddyaboulton — APPROVED** [review]
Thanks @akx !

**Comment by freddyaboulton** [comment]
Can't update the branch and merge @akx  - please merge in main and let me know so we can merge this in.

**Comment by akx** [comment]
@freddyaboulton Rebased, tests were flaky so I re-pushed, we'll see...

**Comment by freddyaboulton** [comment]
Hi @akx - can you reopen the PR and allow edits from maintainers? That way I can merge in main and get this merged! Otherwise might be tricky to align on timing

**Comment by akx** [comment]
@freddyaboulton Sorry, enabled "allow edits" now and rebased.

**Comment by freddyaboulton** [comment]
Thanks again @akx !

**Comment by abidlabs** [comment]
Thanks @akx for addressing the feedback!

**Pull request merged**